istioctl/pkg/ztunnelconfig/ztunnelconfig.go

```diff
@@ -0,0 +1,486 @@
+	CriticalLevel: "critical",
+	OffLevel:      "off",
+}
+
+var stringToLevel = map[string]Level{
+	"trace":    TraceLevel,
+	"debug":    DebugLevel,
+	"info":     InfoLevel,
+	"warning":  WarningLevel,
+	"warn":     WarningLevel,
+	"error":    ErrorLevel,
+	"critical": CriticalLevel,
+	"off":      OffLevel,
+}
+
+var (
+	loggerLevelString = ""
+	reset             = false
+)
+
+func ztunnelLogLevel(level string) string {
+	switch level {
+	case "warning":
+		return "warn"
+	case "critical":
+		return "error"
+	default:
+		return level
+	}
+}
+
+func logCmd(ctx cli.Context) *cobra.Command {
+	var podNamespace string
+	var podNames []string
+
+	logCmd := &cobra.Command{
+		Use:   "log [<type>/]<name>[.<namespace>]",
+		Short: "Retrieves logging levels of the Envoy in the specified pod",
+		Long:  "Retrieve information about logging levels of the Envoy instance in the specified pod, and update optionally",
+		Example: `  # Retrieve information about logging levels for a given pod from Envoy.
+  istioctl proxy-config log <pod-name[.namespace]>
+
+  # Update levels of the all loggers
+  istioctl proxy-config log <pod-name[.namespace]> --level none
+
+  # Update levels of the specified loggers.
+  istioctl proxy-config log <pod-name[.namespace]> --level http:debug,redis:debug
+
+  # Reset levels of all the loggers to default value (warning).
+  istioctl proxy-config log <pod-name[.namespace]> -r
+`,
+		Aliases: []string{"o"},
+		Args: func(cmd *cobra.Command, args []string) error {
+			if labelSelector == "" && len(args) < 1 {
+				cmd.Println(cmd.UsageString())
+				return fmt.Errorf("log requires pod name or --selector")
+			}
+			if reset && loggerLevelString != "" {
+				cmd.Println(cmd.UsageString())
+				return fmt.Errorf("--level cannot be combined with --reset")
+			}
+			if outputFormat != "" && outputFormat != summaryOutput {
+				return fmt.Errorf("--output is not applicable for this command")
+			}
+			return nil
+		},
+		RunE: func(c *cobra.Command, args []string) error {
+			kubeClient, err := ctx.CLIClient()
+			if err != nil {
+				return err
+			}
+			if labelSelector != "" {
+				if podNames, podNamespace, err = getPodNameBySelector(ctx, kubeClient, labelSelector); err != nil {
+					return err
+				}
+			} else {
+				if podNames, podNamespace, err = getPodNames(ctx, args[0], ctx.Namespace()); err != nil {
+					return err
+				}
+			}
+			for _, pod := range podNames {
+				_, err = setupEnvoyLogConfig(kubeClient, "", pod, podNamespace)
+				if err != nil {
+					return err
+				}
+			}
+
+			destLoggerLevels := map[string]Level{}
+			if reset {
+				// reset logging level to `defaultOutputLevel`, and ignore the `level` option
+				levelString, _ := getLogLevelFromConfigMap(ctx)
+				level, ok := stringToLevel[levelString]
+				if ok {
+					destLoggerLevels[defaultLoggerName] = level
+				} else {
+					log.Warn("unable to get logLevel from ConfigMap istio-sidecar-injector, using default value \"info\" for ztunnel")
```

Comment:
This guy needs fixing.

Reply:
since we're no longer using `getLogLevelFromConfigMap`, wdyt of this:

```
destLoggerLevels := map[string]Level{}
if reset {
    log.Warn("logLevel reset; using default value \"info\" for ztunnel")
    loggerLevelString = "info"
} else if loggerLevelString != "" {
```

Reply:
works for me!